azurelinuxagent/common/dhcp.py

```diff
@@ -80,23 +79,30 @@ def wireserver_route_exists(self):
         This is true when running in a virtual network.
         :return: True if a route to KNOWN_WIRESERVER_IP exists.
         """
-        route_exists=False
+        route_exists = False
         logger.info("test for route to {0}".format(KNOWN_WIRESERVER_IP))
         try:
-            route = shellutil.run_get_output("grep -c {0} /proc/net/route".format(KNOWN_WIRESERVER_IP_ENTRY))
+            # grep -c returns 1 if the count is 0, so ignore the return code
+            route = shellutil.run_get_output(
```

Comment:
Why shell out to grep when python is perfectly capable of opening a file, reading it, and counting the lines that contain a fixed string? Since you actually only care about "0" vs. "greater than 0", doing it yourself within python will be faster than shelling out to grep and reading the entire file.